File: business-management-software-frontend/src/services/ProductService.js
import axios from "axios";

const PRODUCT_API_BASE_URL = "http://localhost:8080/api";

class ProductService {
  createProduct(product) {
    return axios.post(PRODUCT_API_BASE_URL + "/products", product);
  }
  getProducts() {
    return axios.get(PRODUCT_API_BASE_URL + "/products");
  }
  searchProducts(searchTerm) {
    return axios.get(PRODUCT_API_BASE_URL + "/products/search/" + searchTerm);
  }

  getProductById(productId) {
    return axios.get(PRODUCT_API_BASE_URL + "/product/" + productId);
  }
  updateProduct(product, productId) {
    return axios.put(PRODUCT_API_BASE_URL + "/product/" + productId, product);
  }

  deleteProduct(productId) {
    return axios.delete(PRODUCT_API_BASE_URL + "/product/" + productId);
  }

  getProductsByCategory(categoryId) {
    return axios.get(PRODUCT_API_BASE_URL + "/products/category/" + categoryId);
  }
}
export default new ProductService();

File: BusinessManagementSoftwareBackend/src/main/java/com/ali/BusinessManagementSoftwareBackend/services/products/ProductService.java
package com.ali.BusinessManagementSoftwareBackend.services.products;

import com.ali.BusinessManagementSoftwareBackend.dto.ProductDto;

import java.io.IOException;
import java.util.List;

public interface ProductService {


    public boolean createProduct(ProductDto productDto) throws IOException ;

    public List<ProductDto> getProducts();
    public List<ProductDto> getDeactivatedProducts() ;
    public void deleteProduct(Long id);
    public List<ProductDto> searchProducts(String keyword) ;



    public ProductDto getProductById(Long id);
    public void deactivateProduct(Long productId) ;
    public List<ProductDto> getProductsByCategory(Long categoryId) ;
    public boolean updateProduct(Long id, ProductDto productDto);



    }


